File: java/src/main/java/com/sequencing/appchains/UsageExample.java
package com.sequencing.appchains;

import java.io.IOException;
import java.util.HashMap;
import java.util.Map;

import com.sequencing.appchains.AppChains.FileResultValue;
import com.sequencing.appchains.AppChains.Report;
import com.sequencing.appchains.AppChains.Result;
import com.sequencing.appchains.AppChains.ResultType;
import com.sequencing.appchains.AppChains.TextResultValue;

public class UsageExample
{
	// Main method for testing purposes
	
	public static void main(String[] args) throws IOException
	{
		AppChains chains = new AppChains("<your token goes here>", "api.sequencing.com");
		
		/**
		 * Low level method invocation example 
		 */
		
		Map<String, Object> rawReport = chains.getRawReport("StartApp", "Chain9", "227680");
		printRawResponse(rawReport);
		
		/**
		 * High level method invocation example
		 */
		
		Report result = chains.getReport("StartApp", "Chain11", "227680");
		printReport(result);

		Map<String, String> appChainsParams = new HashMap<String, String>();
		appChainsParams.put("Chain9",  "227680");
		appChainsParams.put("Chain88",  "227680");
		Map<String, Report> reportMap = chains.getReportBatch("StartAppBatch", appChainsParams);
		printReport(reportMap.get("Chain9"));
		printReport(reportMap.get("Chain88"));

	}
	
	private static void printRawResponse(Map<String, Object> rawReport)
	{
		System.out.println(rawReport);
	}
	
	private static void printReport(Report result) throws IOException
	{
		if (result.isSucceeded() == false)
			System.out.println("Request has failed");
		else
			System.out.println("Request has succeeded");
		
		for (Result r : result.getResults())
		{
			ResultType type = r.getValue().getType();
			
			if (type == ResultType.TEXT)
			{
				TextResultValue v = (TextResultValue) r.getValue();
				System.out.println(String.format(" -> text result type %s = %s", r.getName(), v.getData()));
			}
			
			if (type == ResultType.FILE)
			{
				FileResultValue v = (FileResultValue) r.getValue();
				System.out.println(String.format(" -> file result type %s = %s", r.getName(), v.getUrl()));
				v.saveTo("/tmp");
			}
		}
	}
}

File: cs/Sequencing.AppChainsSample/UsageExample.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Sequencing.AppChainsSample
{
    public class UsageExample
    {
        private static void Main(string[] args)

        {          
            var chains = new AppChains("<your token goes here>", "https://api.sequencing.com/v2",
                "https://beacon.sequencing.com/");

            //Low level method invocation example 
            AppResultsHolder rawReport = chains.GetRawReport("Chain9", "80599");
            PrintRawResponse(rawReport);

            //High level method invocation example
            Report result = chains.GetReport("Chain9", "80599");
            PrintReport("<your token goes here>", result);
            Console.WriteLine("Press any key");
            Console.ReadKey();
        }

        private static void PrintRawResponse(AppResultsHolder rawReport)
        {
            Console.WriteLine(rawReport);
        }

        private static void PrintReport(string token, Report result)
        {
            if (result.Succeeded == false)
                Console.WriteLine("Request has failed");
            else
                Console.WriteLine("Request has succeeded");

            foreach (Result r in result.getResults())
            {
                ResultType type = r.getValue().getType();

                if (type == ResultType.TEXT)
                {
                    var v = (TextResultValue)r.getValue();
                    Console.WriteLine(" -> text result type {0} = {1}", r.getName(), v.Data);
                }

                if (type == ResultType.FILE)
                {
                    var v = (FileResultValue)r.getValue();
                    Console.WriteLine(" -> file result type {0} = {1}", r.getName(), v.Url);
                    v.saveTo(token, ".\\");
                }
            }
        }
    }
}
